x/margin/keeper/keeper_test.go

```diff
@@ -0,0 +1,585 @@
+package keeper_test
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/Sifchain/sifnode/app"
+	clptypes "github.com/Sifchain/sifnode/x/clp/types"
+	"github.com/Sifchain/sifnode/x/margin/test"
+	"github.com/Sifchain/sifnode/x/margin/types"
+	tokenregistrytypes "github.com/Sifchain/sifnode/x/tokenregistry/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestKeeper_Errors(t *testing.T) {
+	_, app := test.CreateTestAppMargin(false)
+	marginKeeper := app.MarginKeeper
+	assert.NotNil(t, marginKeeper)
+}
+
+func TestKeeper_SetMTP(t *testing.T) {
+	t.Run("missed defining asset", func(t *testing.T) {
+		ctx, _, marginKeeper := initKeeper(t)
+		mtp := types.MTP{}
+		err := marginKeeper.SetMTP(ctx, &mtp)
+		assert.EqualError(t, err, "no asset specified: mtp invalid")
+	})
+	t.Run("define asset but no address", func(t *testing.T) {
+		ctx, _, marginKeeper := initKeeper(t)
+		mtp := types.MTP{CollateralAsset: "xxx"}
+		err := marginKeeper.SetMTP(ctx, &mtp)
+		assert.EqualError(t, err, "no address specified: mtp invalid")
+	})
+	t.Run("define asset and address", func(t *testing.T) {
+		ctx, _, marginKeeper := initKeeper(t)
+		mtp := types.MTP{CollateralAsset: "xxx", Address: "xxx"}
+		err := marginKeeper.SetMTP(ctx, &mtp)
+		assert.NoError(t, err)
+	})
+}
+
+func TestKeeper_GetMTP(t *testing.T) {
+	t.Run("get MTP from a store key that exists", func(t *testing.T) {
+		ctx, app, marginKeeper := initKeeper(t)
+		want := addMTPKey(t, ctx, app, marginKeeper, "ceth", "xxx")
+		got, err := marginKeeper.GetMTP(ctx, "ceth", "xxx")
+
+		if err != nil {
+			t.Error("got an error but didn't want one")
+		}
+		if !reflect.DeepEqual(got, want) {
+			t.Errorf("got %v want %v", got, want)
+		}
+	})
+
+	t.Run("fails when store keys does not exist", func(t *testing.T) {
+		ctx, _, marginKeeper := initKeeper(t)
+		marginKeeper.GetMTP(ctx, "ceth", "xxx")
+	})
+}
+
+func TestKeeper_GetMTPIterator(t *testing.T) {
+	ctx, app, marginKeeper := initKeeper(t)
+	want := addMTPKey(t, ctx, app, marginKeeper, "ceth", "xxx")
+	iterator := marginKeeper.GetMTPIterator(ctx)
+	bytesValue := iterator.Value()
+	var got types.MTP
+	types.ModuleCdc.MustUnmarshal(bytesValue, &got)
+
+	if !reflect.DeepEqual(got, want) {
```

Comment:
Don't see a need for deep equal here, I would suggest sticking with the assertion library require.Equal()